### PR TITLE
Fix bool scope config and update throws and update use clases

### PR DESCRIPTION
**Pull request by hryvinskyi**
### Description
Fix bool scope config
Update throws
Update use class

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @hryvinskyi. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by ihor-sviziev** [comment]
Hi @hryvinskyi,
Please look at failed tests and adjust tests or your code.

**Comment by hryvinskyi** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @hryvinskyi. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @hryvinskyi, here is your [new Magento instance](https://pr-18065.engcom.dev.magento.com).
Admin access: [https://pr-18065.engcom.dev.magento.com/admin](https://pr-18065.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`

File: app/code/Magento/AdvancedSearch/Model/Recommendations/DataProvider.php [hunk 1]
```diff
@@ -69,18 +74,24 @@ public function __construct(
...
-        return (bool)$this->scopeConfig->getValue(
+        return $this->scopeConfig->isSetFlag(
             self::CONFIG_RESULTS_COUNT_ENABLED,
             ScopeInterface::SCOPE_STORE
         );
     }
 
     /**
-     * {@inheritdoc}
+     * Get Items
```
Comment (slavvka) [inline app/code/Magento/AdvancedSearch/Model/Recommendations/DataProvider.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/CatalogInventory/Model/Configuration.php [hunk 2]
```diff
@@ -149,7 +152,9 @@ public function __construct(
     }
 
     /**
-     * {@inheritdoc}
+     * Default Scope Id
```
Comment (slavvka) [inline app/code/Magento/CatalogInventory/Model/Configuration.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Elasticsearch/Model/DataProvider/Suggestions.php [hunk 3]
```diff
@@ -90,11 +95,15 @@ public function __construct(
     }
 
     /**
-     * {@inheritdoc}
+     * Get Items
```
Comment (slavvka) [inline app/code/Magento/Elasticsearch/Model/DataProvider/Suggestions.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Elasticsearch/Model/DataProvider/Suggestions.php [hunk 4]
```diff
@@ -118,19 +127,25 @@ public function getItems(QueryInterface $query, $limit = null, $additionalFilter
     }
 
     /**
-     * {@inheritdoc}
+     * Is Results Count Enabled
```
Comment (slavvka) [inline app/code/Magento/Elasticsearch/Model/DataProvider/Suggestions.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/GiftMessage/Model/GiftMessageConfigProvider.php [hunk 5]
```diff
@@ -87,28 +95,31 @@ public function __construct(
     }
 
     /**
-     * {@inheritdoc}
+     * Get Config
```
Comment (slavvka) [inline app/code/Magento/GiftMessage/Model/GiftMessageConfigProvider.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/NewRelicReporting/Model/Config.php [hunk 6]
```diff
@@ -91,7 +91,7 @@ public function __construct(
      */
     public function isNewRelicEnabled()
     {
-        return (bool)$this->scopeConfig->getValue('newrelicreporting/general/enable');
+        return $this->scopeConfig->getValue('newrelicreporting/general/enable');
```
Comment (slavvka) [inline app/code/Magento/NewRelicReporting/Model/Config.php]: Why did you just remove type casting and leave the method `getValue` as is?

File: app/code/Magento/NewRelicReporting/Model/Config.php [hunk 7]
```diff
@@ -181,7 +181,7 @@ public function isSeparateApps()
      */
     public function isCronEnabled()
     {
-        return (bool)$this->scopeConfig->getValue('newrelicreporting/cron/enable_cron');
+        return $this->scopeConfig->getValue('newrelicreporting/cron/enable_cron');
```
Comment (slavvka) [inline app/code/Magento/NewRelicReporting/Model/Config.php]: The same

File: app/code/Magento/Store/Model/HeaderProvider/UpgradeInsecure.php [hunk 8]
```diff
@@ -41,11 +43,13 @@ public function __construct(\Magento\Framework\App\Config\ScopeConfigInterface $
     }
 
     /**
-     * {@inheritdoc}
+     * Whether the header should be attached to the response
```
Comment (slavvka) [inline app/code/Magento/Store/Model/HeaderProvider/UpgradeInsecure.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/HeaderProvider/Hsts.php [hunk 9]
```diff
@@ -41,11 +43,13 @@ public function __construct(\Magento\Framework\App\Config\ScopeConfigInterface $
     }
 
     /**
-     * {@inheritdoc}
+     * Whether the header should be attached to the response
```
Comment (slavvka) [inline app/code/Magento/Store/Model/HeaderProvider/Hsts.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 10]
```diff
@@ -114,23 +117,29 @@ public function __construct(
     }
 
     /**
-     * {@inheritdoc}
+     * Set current default store
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 11]
```diff
@@ -114,23 +117,29 @@ public function __construct(
...
+     * @param string $store
      */
     public function setCurrentStore($store)
     {
         $this->currentStoreId = $store;
     }
 
     /**
-     * {@inheritdoc}
+     * Allow or disallow single store mode
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: The same

File: app/code/Magento/Wishlist/Model/Rss/Wishlist.php [hunk 12]
```diff
@@ -272,7 +283,9 @@ public function getFeeds()
     }
 
     /**
-     * {@inheritdoc}
+     * Is Auth Required
```
Comment (slavvka) [inline app/code/Magento/Wishlist/Model/Rss/Wishlist.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: lib/internal/Magento/Framework/Session/SidResolver.php [hunk 13]
```diff
@@ -169,7 +176,7 @@ public function getUseSessionInUrl()
         if ($this->_useSessionInUrl === null) {
             //Using config value by default, can be overridden by using the
             //setter.
-            $this->_useSessionInUrl = (bool)$this->scopeConfig->getValue(
+            $this->_useSessionInUrl = $this->scopeConfig->getValue(
```
Comment (slavvka) [inline lib/internal/Magento/Framework/Session/SidResolver.php]: The same. You removed type casting and left the method intact

**Review by slavvka — CHANGES_REQUESTED** [review]
Please fix the comments and other similar places

**Comment by hryvinskyi** [comment]
@slavvka done

File: app/code/Magento/Customer/Ui/Component/DataProvider/Document.php [hunk 14]
```diff
@@ -93,7 +94,12 @@ public function __construct(
     }
 
     /**
-     * @inheritdoc
```
Comment (slavvka) [inline app/code/Magento/Customer/Ui/Component/DataProvider/Document.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Customer/Ui/Component/DataProvider/Document.php [hunk 15]
```diff
@@ -93,7 +94,12 @@ public function __construct(
...
+     * Get Custom Attribute
```
Comment (slavvka) [inline app/code/Magento/Customer/Ui/Component/DataProvider/Document.php]: You probably missed that

File: app/code/Magento/Elasticsearch/Model/DataProvider/Suggestions.php (hunk 4 above)
Comment (slavvka) [reply app/code/Magento/Elasticsearch/Model/DataProvider/Suggestions.php]: You probably missed that

File: app/code/Magento/Store/Model/StoreManager.php [hunk 16]
```diff
@@ -114,23 +117,25 @@ public function __construct(
...
+     * @inheritdoc
      */
     public function setIsSingleStoreModeAllowed($value)
     {
         $this->isSingleStoreAllowed = $value;
     }
 
     /**
-     * {@inheritdoc}
+     * Check if store has only one store view
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 17]
```diff
@@ -139,15 +144,21 @@ public function hasSingleStore()
     }
 
     /**
-     * {@inheritdoc}
+     * Check if system is run in the single store mode
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 18]
```diff
@@ -139,15 +144,21 @@ public function hasSingleStore()
...
+     * @return bool
      */
     public function isSingleStoreMode()
     {
         return $this->isSingleStoreModeEnabled() && $this->hasSingleStore();
     }
 
     /**
-     * {@inheritdoc}
+     * Retrieve application store object
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 19]
```diff
@@ -171,7 +182,11 @@ public function getStore($storeId = null)
     }
 
     /**
-     * {@inheritdoc}
+     * Retrieve stores array
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 20]
```diff
@@ -190,7 +205,11 @@ public function getStores($withDefault = false, $codeKey = false)
     }
 
     /**
-     * {@inheritdoc}
+     * Retrieve application website object
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 21]
```diff
@@ -210,7 +229,11 @@ public function getWebsite($websiteId = null)
     }
 
     /**
-     * {@inheritdoc}
+     * Get loaded websites
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 22]
```diff
@@ -229,7 +252,9 @@ public function getWebsites($withDefault = false, $codeKey = false)
     }
 
     /**
-     * {@inheritdoc}
+     * Reinitialize store list
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 23]
```diff
@@ -242,7 +267,10 @@ public function reinitStores()
     }
 
     /**
-     * {@inheritdoc}
+     * Retrieve default store for default group and website
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 24]
```diff
@@ -252,7 +280,12 @@ public function getDefaultStoreView()
     }
 
     /**
-     * {@inheritdoc}
+     * Retrieve application store group object
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 25]
```diff
@@ -267,7 +300,10 @@ public function getGroup($groupId = null)
     }
 
     /**
-     * {@inheritdoc}
+     * Prepare array of store groups
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

File: app/code/Magento/Store/Model/StoreManager.php [hunk 26]
```diff
@@ -307,7 +345,11 @@ private function getStoreWebsiteRelation()
     }
 
     /**
-     * @inheritdoc
+     * Get assigned to website store
```
Comment (slavvka) [inline app/code/Magento/Store/Model/StoreManager.php]: Please revert the comment and just remove curly braces `{}`. Please refer to the guide https://devdocs.magento.com/guides/v2.3/coding-standards/docblock-standard-general.html#inheritdoc

**Review by slavvka — CHANGES_REQUESTED** [review]
You probably missed some of my comments. Also there are the same changes that should be fixed. Please look through all your changes thoroughly and fix all them. Thank you!

**Comment by hryvinskyi** [comment]
@slavvka done

**Comment by slavvka** [comment]
@hryvinskyi Thank you! Let's wait for the tests

**Comment by slavvka** [comment]
hey @hryvinskyi you still have failing tests. Also please merge the latest mainline code and resolve the conflict.

**Review by slavvka — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @slavvka, thank you for the review.
[ENGCOM-3164](https://jira.corp.magento.com/browse/ENGCOM-3164) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @hryvinskyi. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Comment by slavvka** [comment]
@hryvinskyi thank you. Could you please consider porting this PR to 2.2? Yuo can use the automatic tool https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#porting-tool

**Pull request merged**